cdap-authorization-dataset-plugin/src/main/java/co/cask/cdap/security/authorization/DatasetBasedAuthorizer.java

```diff
@@ -67,11 +71,22 @@ public ACLDataset get() {
         return new ACLDataset(table);
       }
     };
+    Properties properties = context.getExtensionProperties();
+    if (properties.containsKey("superusers")) {
+      for (String superuser : Splitter.on(",").trimResults().omitEmptyStrings()
+        .split(properties.getProperty("superusers"))) {
+        superUsers.add(new Principal(superuser, Principal.PrincipalType.USER));
+      }
+    }
   }
 
   @Override
   public void enforce(final EntityId entity, final Principal principal,
                       final Action action) throws UnauthorizedException, TransactionFailureException {
+    // no enforcement for superusers
+    if (superUsers.contains(principal)) {
```

Comment:
So we thought about this. The idea is - ``superUsers`` is just one way of bootstrapping authorization. Each extension (Sentry, Ranger, etc) could have their own way of bootstrapping. e.g. Instead of bootstrapping by way of ``superUsers``, an extension could expect a file that has some default privileges set for bootstrapping. Thoughts?